usl_pipeline/usl_lib/usl_lib/shared/suppliers.py

```diff
@@ -0,0 +1,49 @@
+import abc
```

Comment:
Might just stick this code alongside the chunking code rather than create a new module for it as I wouldn't expect it to be used in any other contexts. This code and the chunking code feel pretty entwined.

What else do you anticipate adding to this new 'suppliers' module? Would it expand past its current contents?

Is there a more clear name that 'suppliers' you could imagine? It looks like this is more like a `path` or `path_utils` or such?

What does the 'shared' part of `usl_lib/shared/suppliers` indicate? I think of everything in 'usl_lib' as being shared -- 'lib' already implies 'shared' to me.

Reply:
It's supposed to be shared between different chunkers. But it's not relevant anymore.